src/Build/BackEnd/TaskExecutionHost/TaskExecutionHost.cs

```diff
@@ -1009,8 +997,8 @@ private ITask InstantiateTask(int scheduledNodeId, IDictionary<string, string> t
                         else
                         {
                             // Normal in-process execution for custom task factories
-                            task = _taskFactoryWrapper.TaskFactory is ITaskFactory2 taskFactory2 ?
-                                taskFactory2.CreateTask(taskFactoryEngineContext, taskIdentityParameters) :
+                            task = _taskFactoryWrapper.TaskFactory is ITaskFactory3 taskFactory3 ?
+                                taskFactory3.CreateTask(taskFactoryEngineContext, taskIdentityParameters) :
```

Comment:
This too feels like it should have a compat shim for `ITaskFactory2` implementations.

Reply:
added